rust-manifest/src/errors.rs

```diff
@@ -13,6 +14,25 @@ pub enum Error {
     UnsupportedVersion(String),
 }
 
+impl error::Error for Error {
+    fn description(&self) -> &str {
+        use self::Error::*;
+        match *self {
+            Parsing(_) => "error parsing manifest",
+            MissingKey(_) => "missing key",
+            ExpectedType(_, _) => "expected type",
+            PackageNotFound(_) => "package not found",
+            TargetNotFound(_) => "target not found",
+            MissingRoot => "manifest has no root package",
+            UnsupportedVersion(_) => "unsupported manifest version",
+        }
+    }
+
+    fn cause(&self) -> Option<&error::Error> {
+        None
```

Comment:
there's an opportunity here to return the first parse error for the `Parsing` variant. easily done if you think that makes sense.